GeoLib/PointVec.h

```diff
@@ -143,6 +145,10 @@ class PointVec : public TemplateVec<Point>
 	 */
 	std::vector<std::size_t> _pnt_id_map;
 
+	/// The reverse map to the name to id map, for fast lookup of the name to a
+	/// given point id.
+	std::vector<std::string> _id_to_name_map;
```

Comment:
The mapping "name -> item" is already there. It is implemented employing a `std::map<std::string, std::size_t>`. Sometimes the reverse lookup is needed (item or id -> name), for instance, in the case when all points should be written to a file. Then only the id of the point is given and a name is searched. Currently this name search is very slow. The changes in the PR try to speedup the name search.

You are right: this takes more memory. But the new implemented `std::vector<std::string>` allows a faster access (hours vs. seconds).